magefiles/installation/oauth.go

```diff
@@ -0,0 +1,120 @@
+)
+
+const (
+	HTPASSWD_FILE_NAME      = "ci.htpaswd"
+	DEFAULT_IDP_SECRET_NAME = "htpass-secret"
+	DEFAULT_IDP_NAME        = "ci_identity"
+	DEFAULT_OAUTH_NAME      = "cluster"
+	DEFAULT_HTPASSWD_BIN    = "htpasswd"
+)
+
+var (
+	randomOCPUserPass = util.GenerateRandomString(10)
+	randomOCPUserName = util.GenerateRandomString(6)
+	httpaswdArgs      = []string{"-c", "-B", "-b", HTPASSWD_FILE_NAME, randomOCPUserName, randomOCPUserPass}
+)
+
+// CreateOauth generate a new random admin user for testing. This functionality will be executed only in Openshift CI.
+func (i *InstallAppStudio) CreateOauth() error {
+	if os.Getenv("CI") != "true" {
+		return nil
+	}
+
+	if err := i.generateHttpaswd(); err != nil {
+		return err
+	}
+
+	if err := utils.ExecuteCommandInASpecificDirectory("oc", []string{"create", "secret", "generic", DEFAULT_IDP_SECRET_NAME, fmt.Sprintf("--from-file=htpasswd=%s/%s", i.TmpDirectory, HTPASSWD_FILE_NAME), "-n", "openshift-config"}, ""); err != nil {
+		return err
+	}
+
+	if err := i.updateOauthCluster(); err != nil {
+		return err
+	}
+	return i.loginAsNewUser()
+}
+
+// updateOauthCluster update the existing oauth object in the openshift cluster
+func (i *InstallAppStudio) updateOauthCluster() error {
+	namespacedName := types.NamespacedName{
+		Name: DEFAULT_OAUTH_NAME,
+	}
+
+	oauth := &ocpOauth.OAuth{}
+	if err := i.KubernetesClient.KubeRest().Get(context.TODO(), namespacedName, oauth); err != nil {
+		return err
+	}
+
+	updateObj := &ocpOauth.OAuth{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: ocpOauth.GroupVersion.Version,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: DEFAULT_OAUTH_NAME,
+		},
+		Spec: ocpOauth.OAuthSpec{
+			IdentityProviders: []ocpOauth.IdentityProvider{
+				{
+					MappingMethod: ocpOauth.MappingMethodClaim,
+					Name:          DEFAULT_IDP_NAME,
+					IdentityProviderConfig: ocpOauth.IdentityProviderConfig{
+
+						Type: ocpOauth.IdentityProviderTypeHTPasswd,
+						HTPasswd: &ocpOauth.HTPasswdIdentityProvider{
+							FileData: ocpOauth.SecretNameReference{
+								Name: DEFAULT_IDP_SECRET_NAME,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	updateObj.SetResourceVersion(oauth.GetResourceVersion())
+
+	return i.KubernetesClient.KubeRest().Update(context.Background(), updateObj)
+}
+
+// generateHttpaswd create a new htpasswd file with random user and password
+func (i *InstallAppStudio) generateHttpaswd() error {
+	return utils.ExecuteCommandInASpecificDirectory(DEFAULT_HTPASSWD_BIN, httpaswdArgs, i.TmpDirectory)
+}
+
+// loginAsNewUser add cluster-admin role to a random user and then generate a new kubeconfig and login to the cluster. This func will be executed with an openshift admin user. In openshift CI
+// by default the admin user is system:admin
+func (i *InstallAppStudio) loginAsNewUser() error {
+	if err := utils.ExecuteCommandInASpecificDirectory("oc", []string{"adm", "policy", "add-cluster-role-to-user", "cluster-admin", randomOCPUserName}, ""); err != nil {
```

Comment:
similar question about using the golang library

Reply:
TBH I would like to avoid importing oc cli in our go.mod. We will start to have to much dependency and the cli is a big one.

Reply:
[we're already importing that dependency](https://github.com/redhat-appstudio/e2e-tests/blob/cc1b8680950f04f194f857160d0f0474e4fe7034/go.mod#L21) ;)

Reply:
https://github.com/openshift/oc/blob/619c8d74223a1b21fb182e11c35592716aed2896/go.mod#L188
No, we are importing only the cli/image
This is another replace to add to our code  and i would like to avoid adding.